ckeditor/ckeditor5
# Split all snippets into smaller chunks while building the docs

### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Internal: Split all snippets into smaller chunks while building the docs.

---

### Additional information

**Verification scenario**

1. Before switching to the `ci/1882` branch, change the `--max-old-space-size` memory limit for the `docs` script in the `package.json` file in CKEditor 5 from `8192` to `4096`.
1. Run `yarn run docs --production`.
   - In the middle of building the docs, there is an unexpected out of memory error.
1. Switch to the `ci/1882` branch.
1. Run `yarn run docs --production` again.
1. Confirm that the docs are built successfully without any error.
   - You can check the docs by the crawler: `yarn run docs:serve` and `yarn run docs:verify` in new terminal.
 
      <img src="https://user-images.githubusercontent.com/56868128/182346538-49098360-f76b-4bdb-8cd0-65579e7d3d50.png" width="280px">
 
   - You can also check some guides manually and confirm that the snippets work as expected.

**More thorough verification scenario**

1. Change `SNIPPETS_BUILD_CHUNK_SIZE` to `10`.
1. Add `console.log( webpackConfigs );` in [this line](https://github.com/ckeditor/ckeditor5/pull/12204/files#diff-64df00835cabfa8b0cb333f7da1cda6a0abfe60968806d6c6f51c6722db102d9R132).
1. Run `yarn run docs --production`.
1. In the printed log you can check how many separate webpack processes will be used - each webpack config (containing own `entry` list) will be built separately from the others. Although you have set the limit to 10, some configs contain more than 10 entries, e.g. the config containing the snippets from the table guide has 15 entries, because it is not possible to split in the middle the table snippets belonging to the same page.
